hazelcast/src/main/java/com/hazelcast/client/impl/protocol/util/ParameterUtil.java

```diff
@@ -32,11 +32,11 @@ public static int calculateDataSize(String string) {
     }
 
     public static int calculateDataSize(Data data) {
-        return calculateDataSize(data.toByteArray());
+        return calculateDataSize(data.totalSize());
     }
 
     public static int calculateDataSize(Map.Entry<Data, Data> entry) {
-        return calculateDataSize(entry.getKey().toByteArray()) + calculateDataSize(entry.getValue().toByteArray());
+        return calculateDataSize(entry.getKey().totalSize()) + calculateDataSize(entry.getValue().totalSize());
```

Comment:
Same typo here.
`return entry.getKey().totalSize() + entry.getValue().totalSize()`
And you need to take size field into account . You need to add Bits.INT_SIZE_IN_BYTES per data .
As we were doing in here when byte[] is used to calculate data size
```
    public static int calculateDataSize(byte[] bytes) {
        return Bits.INT_SIZE_IN_BYTES + bytes.length;
    }
```